### ISSUE TITLE
pytest testcase with @pytest.mark.asyncio still run in sync mode but not async mode

### ISSUE THREAD

**Issue by SilenceHe** [pytest-dev/pytest-asyncio#206]
why pytest testcase with @pytest.mark.asyncio still run in sync mode but not async mode?

I have testcase with pytest like below:

**test_tmp.py**

```python
import logging
import pytest
import asyncio
import time


@pytest.mark.asyncio
async def test_async_01():
    logging.info(f"time => {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time()))}")
    await asyncio.sleep(5)


@pytest.mark.asyncio
async def test_async_02():
    logging.info(f"time => {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time()))}")
    await asyncio.sleep(5)
```

When I try to run this test module with command **pytest test_tmp.py**, what I expect is these two cases run almost at the same time as its **@pytest.mark.asyncio** marked, should run with **async** mode, but in fact, it looks run one by one with **sync** mode, is there something wrong with my code? or how to do to make it works as expect?

**NOTE: I didn't override event_loop fixture, it should used the default one.**

**The related screenshot like below:**
![image](https://user-images.githubusercontent.com/4414171/114264324-bc122680-9a1c-11eb-99ba-e8652fc54d0f.png)

**Comment by denolehov** [comment]
This plugin doesn't make `pytest` itself run in async context, it simply provides an event loop so each _individual_ test can be executed inside of a temporary loop that has been created for each `async` test.

In other words, tests are executed synchronously, but each async test has its own event loop to be executed in, thanks to this plugin.

**Comment by heliping** [comment]
when I use this function to webUI test， only the last testcase can be success, the first one already lost the mouse.

**Comment by seifertm** [comment]
@SilenceHe your is absolutely correct. This is the intended behavior. as @denolehov explained the individual test cases do not run concurrently. This is by design and it doesn't look like it's going to change anytime soon. Pytest-asyncio only allows you to make `async def` test cases and to call `await` in test functions. 

If you need things to run in parallel within the same test case, you need to use the regular asyncio means, for example:
```
import asyncio
import pytest

@pytest.mark.asyncio
async def test_fixture(request):
    coro1 = asyncio.sleep(5)
    coro2 = asyncio.sleep(5)
    await asyncio.gather(coro1, coro2)
```

This test case finishes in just over five seconds, even though both coroutines sleep for five seconds each:
```
$ pytest
============ test session starts ============
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /tmp/test
plugins: hypothesis-6.34.2, asyncio-0.16.0
collected 1 item                                                                                                                    

test_example.py .                                                                                                             [100%]

=========== 1 passed in 5.02s ============
```

@SilenceHe I am closing this as `wontfix`, because the behavior you are showing is expected. Please feel free to continue the discussion, if you cannot make pytest-asyncio work the way you need it.

@heliping I am sorry, but I don't see how your comment relates to the issue of the OP. Could you open a separate issue for your problem?
